DeleteEycaRemoteActivity/handler.ts

```diff
@@ -18,13 +18,14 @@ export const ActivityInput = t.interface({
 export type ActivityInput = t.TypeOf<typeof ActivityInput>;
 
 export const getDeleteEycaRemoteActivityHandler = (
-  redisClient: RedisClient,
+  redisClientPromise: Promise<RedisClient>,
```

Comment:
Don't pass a promise as a value, it's problematic ancd can lead to unexpected behaviour (a promise can only be resolved once).

Instead, pass a lazy promise.

---

I see the point why you're doing that, I think it would be better implemented via a lazy-initialized singleton

Reply:
Thank you, I was just thinking about that, I thought that a review would have suggested something better.

Reply:
@balanza it should be better in https://github.com/pagopa/io-functions-cgn/pull/102/commits/449f2df8e7efd03494c4f8b7f5e86da23c842a6f what do you think?